packages/camera/camera_avfoundation/ios/camera_avfoundation/Sources/camera_avfoundation/CameraPlugin.swift

```diff
@@ -0,0 +1,528 @@
+        } else {
+          strongSelf.createCameraOnSessionQueue(
+            withName: cameraName,
+            settings: settings,
+            completion: completion)
+        }
+      }
+    }
+  }
+
+  public func createCameraOnSessionQueue(
+    withName: String,
+    settings: FCPPlatformMediaSettings,
+    completion: @escaping (NSNumber?, FlutterError?) -> Void
+  ) {
+    captureSessionQueue.async { [weak self] in
+      self?.sessionQueueCreateCamera(name: withName, settings: settings, completion: completion)
+    }
+  }
+
+  // This must be called on captureSessionQueue. It is extracted from
+  // initializeCamera:withImageFormat:completion: to make it easier to reason about strong/weak
+  // self pointers.
+  private func sessionQueueCreateCamera(
+    name: String,
+    settings: FCPPlatformMediaSettings,
+    completion: @escaping (NSNumber?, FlutterError?) -> Void
+  ) {
+    let mediaSettingsAVWrapper = FLTCamMediaSettingsAVWrapper()
+
+    let camConfiguration = FLTCamConfiguration(
+      mediaSettings: settings,
+      mediaSettingsWrapper: mediaSettingsAVWrapper,
+      captureDeviceFactory: { [self] in
+        self.captureDeviceFactory(name)
+      },
+      captureSessionFactory: captureSessionFactory,
+      captureSessionQueue: captureSessionQueue,
+      captureDeviceInputFactory: captureDeviceInputFactory
+    )
+
+    var error: NSError?
+    let newCamera = FLTCam(configuration: camConfiguration, error: &error)
+
+    if let error = error {
+      completion(nil, CameraPlugin.flutterErrorFromNSError(error))
+    } else {
+      camera?.close()
+      camera = newCamera
+
+      FLTEnsureToRunOnMainQueue { [weak self] in
+        guard let strongSelf = self else { return }
+        completion(NSNumber(value: strongSelf.registry.register(newCamera)), nil)
+      }
+    }
+  }
+
+  public func initializeCamera(
+    _ cameraId: Int,
+    withImageFormat imageFormat: FCPPlatformImageFormatGroup,
+    completion: @escaping (FlutterError?) -> Void
+  ) {
+    captureSessionQueue.async { [weak self] in
+      self?.sessionQueueInitializeCamera(
+        cameraId,
+        withImageFormat: imageFormat,
+        completion: completion)
+    }
+  }
+
+  // This must be called on captureSessionQueue. It is extracted from
+  // initializeCamera:withImageFormat:completion: to make it easier to reason about strong/weak
+  // self pointers.
+  private func sessionQueueInitializeCamera(
+    _ cameraId: Int,
+    withImageFormat imageFormat: FCPPlatformImageFormatGroup,
+    completion: @escaping (FlutterError?) -> Void
+  ) {
+    camera?.videoFormat = FCPGetPixelFormatForPigeonFormat(imageFormat)
+
+    camera?.onFrameAvailable = { [weak self] in
+      guard let camera = self?.camera else { return }
+      if !camera.isPreviewPaused {
+        FLTEnsureToRunOnMainQueue { [weak self] in
+          self?.registry.textureFrameAvailable(Int64(cameraId))
+        }
+      }
+    }
+
+    camera?.dartAPI = FCPCameraEventApi(
```

Comment:
this is strange, why is camera still optional?

Reply:
You mean regarding the `guard` above? That guard is in the `onFrameAvailable` lambda and this is outside of it. I will add a guard outside